eng/packages/http-client-csharp-mgmt/emitter/src/resource-detection.ts

```diff
@@ -428,9 +428,10 @@ function getResourceScopeOfMethod(
   return undefined;
 }
 
-// TODO -- this logic needs to be refined in the near future.
 function getOperationScope(path: string): ResourceScope {
-  if (
+  if (path.startsWith("/{resourceUri}") || path.startsWith("/{scope}")) {
```

Comment:
seems like extension resource operation should always have `[@extensionResourceOperation](https://github.com/Azure/typespec-azure/blob/a6c4dc8f82a7ab92c631d720a32854aec924fbda/packages/typespec-azure-resource-manager/lib/extension/operations.tsp#L55C2-L55C28)` decorator. Can we use that to detect the extension scope?

Reply:
searched the tspCodeModel.json in test project, there is no @extensionResourceOperation decorator found. If the decorator is provided by upstream, we can definitely using it to detect the scope.

Reply:
I think using path here as a temporary solution might also be acceptable right now - there might not be a way in typespec that people can customize the name of this parameter.
Depending on how long we would like to stay on the current status, because these logics will be replaced by calling the resolveArmResources API, we do not have to do too many changes here if it could work fine.